src/main/java/lotto/controller/PrizeController.java

```diff
@@ -0,0 +1,50 @@
+package lotto.controller;
+
+import lotto.domain.lottery.Lotto;
+import lotto.domain.prize.Prize;
+import lotto.exception.LottoException;
+import lotto.service.Parser;
+import lotto.view.input.InputReader;
+import lotto.view.output.OutputWriter;
+
+import java.util.List;
+
+import static lotto.view.constants.PrintMessage.REQUEST_BONUS_NUMBER;
+import static lotto.view.constants.PrintMessage.REQUEST_PRIZE_NUMBER;
+
+public class PrizeController {
+    private PrizeController() {
+    }
+
+    public static Prize requestPrizeNumbers() {
+        OutputWriter.printMessage(REQUEST_PRIZE_NUMBER);
+        Lotto prizeNumbers = readPrizeNumbers();
+
+        OutputWriter.printNewLine();
+        OutputWriter.printMessage(REQUEST_BONUS_NUMBER);
+        return requestBonusNumber(prizeNumbers);
+    }
+
+    public static Prize requestBonusNumber(final Lotto prizeNumbers) {
+        try {
+            final String bonusNumberInput = InputReader.readLine();
+            final int bonusNumber = Parser.parseStringToInt(bonusNumberInput);
+            return Prize.of(prizeNumbers, bonusNumber);
+        } catch (LottoException exception) {
+            OutputWriter.println(exception.getMessage());
+            return requestBonusNumber(prizeNumbers);
+        }
+    }
+
+    private static Lotto readPrizeNumbers() {
+        try {
+            final String prizeNumbers = InputReader.readLine();
+            List<Integer> numbers = Parser.splitByDelimiter(prizeNumbers);
+
+            return new Lotto(numbers);
+        } catch (LottoException exception) {
+            OutputWriter.println(exception.getMessage());
+            return readPrizeNumbers();
+        }
+    }
+}
```

Comment:
[이렇게 바꿔보시면 좋을 것 같아요!]
try 에서 코드가 실행 후 결과를 반환하고 실패하면 다시 동작한다와 같은 기능이 반복적으로 사용되고 있습니다!
이부분을 다음과 같이 수정하면 재실행 기능과 동작 기능을 분리할 수 있을 것 같아요!
```suggestion
    private static Lotto readPrizeNumbers() {
        return readType1( () -> {
            final String prizeNumbers = InputReader.readLine();
            List<Integer> numbers = Parser.splitByDelimiter(prizeNumbers);
            return new Lotto(numbers);
        });
    }

    private static <T> T readType1(Supplier<T> supplier){
        try{
            return supplier.get();
        }catch (LottoException exception) {
            OutputWriter.println(exception.getMessage());
            return readType1(supplier);
        }
    }
```
함수명은 임의로 지었기 때문에 편하게 바꾸시면 될 것 같습니다!

Reply:
이번에 재홍님 코드 보면서, 입출력 부분을 제네릭과 Supplier로 해결해보려고 시도해보고 있었는데, 이렇게 먼저 친절하게 정리해서 제안해주시니 감사할 따름입니다...언제나 넘치는 지식과 코드 센스에 감탄하며 배우고 있습니다!! 늘 감사합니다!! 👍